cmd/status/cmdstatus_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package status
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/cli-utils/pkg/apply/poller"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling"
+	pollevent "sigs.k8s.io/cli-utils/pkg/kstatus/polling/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/object"
+)
+
+var (
+	inventoryTemplate = `
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  labels:
+    cli-utils.sigs.k8s.io/inventory-id: test
+  name: foo
+  namespace: default
+`
+	depObject = object.ObjMetadata{
+		Name:      "foo",
+		Namespace: "default",
+		GroupKind: schema.GroupKind{
+			Group: "apps",
+			Kind:  "Deployment",
+		},
+	}
+
+	stsObject = object.ObjMetadata{
+		Name:      "bar",
+		Namespace: "default",
+		GroupKind: schema.GroupKind{
+			Group: "apps",
+			Kind:  "StatefulSet",
+		},
+	}
+)
+
+func TestStatusCommand(t *testing.T) {
+	testCases := map[string]struct {
+		pollUntil      string
+		printer        string
+		timeout        time.Duration
+		input          string
+		inventory      []object.ObjMetadata
+		events         []pollevent.Event
+		expectedErrMsg string
+		expectedOutput string
+	}{
+		"no inventory template": {
+			input:          "",
+			expectedErrMsg: "Package uninitialized. Please run \"init\" command.",
```

Comment:
This message is confusing. We should have a wrapper in upstream which parses this message and give clear command to be used.

Reply:
This is the same message we use for apply, and this is not the full error message. It has a more thorough explanation.
I agree we should try to make the instructions on which command to run more clear, but it requires a larger change since it depends on whether this is used by kpt or other tools. I'll revisit that in a separate PR.

Reply:
Agree.